### PR TITLE
Reorganize so that a single package contains all the user's code

### PR DESCRIPTION
**Pull request by pmorie**
Kind of big PR (for this little project) that reorganizes the project so that the user should only need to put code into the `pkg/user` package.

@LiliC I will wait for you to take a look at this before merging, since I know you have changes in progress too.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,4 +3,10 @@
...
 A go quickstart for creating service brokers that implement the
-[Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker).
\ No newline at end of file
+[Open Service Broker API](https://github.com/openservicebrokerapi/servicebroker).
+Broker authors implement an interface that uses the same types as the
+[go-open-service-broker-client](https://github.com/pmorie/go-open-service-broker-client)
+project.
+
+## Who should use this project?
+
```
Comment (pmorie) [inline README.md]: Oops, that was for another branch. Should not pack for a week and try to code at same time

File: pkg/rest/apisurface.go [hunk 2]
```diff
@@ -8,8 +8,13 @@ import (
 	"github.com/gorilla/mux"
 
 	osb "github.com/pmorie/go-open-service-broker-client/v2"
+
+	"github.com/pmorie/go-open-service-broker-skeleton/pkg/broker"
+	"github.com/pmorie/go-open-service-broker-skeleton/pkg/user"
 )
 
+// TODO: union this package with server
```
Comment (LiliC) [inline pkg/rest/apisurface.go]: 👍

File: pkg/rest/util.go [hunk 3]
```diff
@@ -8,6 +8,11 @@ import (
 	"github.com/pmorie/go-open-service-broker-client/v2"
 )
 
+// TODO: export in go-open-service-broker-client
```
Comment (LiliC) [inline pkg/rest/util.go]: 👍 💯

Comment (LiliC) [reply pkg/rest/util.go]: As as well the constants in the above file? :)

File: pkg/server/server.go [hunk 4]
```diff
@@ -9,21 +9,23 @@ import (
 
 	"github.com/golang/glog"
 
-	"github.com/pmorie/go-open-service-broker-skeleton/pkg/broker"
+	"github.com/pmorie/go-open-service-broker-skeleton/pkg/rest"
 )
 
-type server struct {
-	api broker.APISurface
+// Server might be redundant :)
```
Comment (LiliC) [inline pkg/server/server.go]: Yeah I agree, took the scaffolding code from me UPS broker.

Comment (pmorie) [reply pkg/server/server.go]: Yeah, that was a note to myself to collapse them but I ran out of time.

**Pull request merged**